snowpack/src/plugins/plugin-esbuild.ts

```diff
@@ -30,11 +30,17 @@ export function esbuildPlugin(config: SnowpackConfig, {input}: {input: string[]}
     async load({filePath}) {
       esbuildService = esbuildService || (await startService());
       const contents = await fs.readFile(filePath, 'utf-8');
+      let jsxFactory = config.buildOptions.jsxFactory;
+      let jsxFragment = config.buildOptions.jsxFragment;
       const isPreact = checkIsPreact(filePath, contents);
+      if (isPreact) {
+        jsxFactory || (jsxFactory = 'h')
```

Comment:
This is a bit different from how we treat conditional variable setting in the rest of the codebase. for consistency, please do something closer to this:

```js
const jsxFactory = config.buildOptions.jsxFactory ?? (isPreact ? 'h' : undefined);
const jsxFragment = config.buildOptions. jsxFragment ?? (isPreact ? 'Fragment' : undefined);
```